### PR TITLE
[WIP] Update Dockerfiles

### PR DESCRIPTION
**Pull request by angrybayblade**
## Proposed changes

This PR updated dockerfile as proposed in the deployment flow doc

## Types of changes

What types of changes does your code introduce? (A **breaking change** is a fix or feature that would cause existing functionality and APIs to not work as expected.)
_Put an `x` in the box that applies_

- [ ] Non-breaking fix (non-breaking change which fixes an issue)
- [ ] Breaking fix (breaking change which fixes an issue)
- [ ] Non-breaking feature (non-breaking change which adds functionality)
- [ ] Breaking feature (breaking change which adds functionality)
- [ ] Refactor (non-breaking change which changes implementation)
- [ ] Messy (mixture of the above - requires explanation!)

## Checklist

_Put an `x` in the boxes that apply._

- [x] I have read the [CONTRIBUTING](../blob/main/CONTRIBUTING.md) doc
- [x] I am making a pull request against the `main` branch (left side). Also you should start your branch off our `main`.
- [x] Lint and unit tests pass locally with my changes
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] I have locally run services that could be impacted and they do not present failures derived from my changes

### CODE REVIEW

File: tests/test_autonomy/test_replay.py [hunk 1]
```diff
@@ -49,7 +49,7 @@
     "environment": [
         "LOG_FILE=/logs/aea_0.txt",
         "ID=0",
-        "VALORY_APPLICATION=valory/oracle:0.1.0:QmUTuBXUbgBjBbiLR3kmBxZXQowseu2mc6xR5MVuX8UyXJ",
+        "AGENT_ID=valory/oracle:0.1.0:QmUTuBXUbgBjBbiLR3kmBxZXQowseu2mc6xR5MVuX8UyXJ",
```
Comment (DavidMinarsch) [inline tests/test_autonomy/test_replay.py:52]: Oh yes, thank you  - let's not introduce unnecessary terminology

File: oracle_hardhat/README.md [hunk 2]
```diff
@@ -0,0 +1,3 @@
+## BTC Price Oracle (Hardhat network)
```
Comment (DavidMinarsch) [inline oracle_hardhat/README.md:1]: Hm on main we already have oracle_hardhat service in packages - accidental commit?

File: hello_world/service.yaml [hunk 3]
```diff
@@ -0,0 +1,40 @@
+name: hello_world
```
Comment (DavidMinarsch) [inline hello_world/service.yaml:1]: Accidental?

**Review by DavidMinarsch — COMMENTED** [review]
Ugh why are you moving things back into a deployments folder?

File: deployments/Dockerfiles/autonomy/Dockerfile [hunk 4]
```diff
@@ -1,34 +1,26 @@
-ARG AEA_VERSION=latest
-FROM valory/open-aea-user:${AEA_VERSION}
-ARG AUTHOR=default_author
+FROM valory/open-aea-user:latest
 ARG AEA_AGENT
```
Comment (DavidMinarsch) [inline deployments/Dockerfiles/autonomy/Dockerfile:2]: why the reorg?, why the removal of version and author?

File: autonomy/deploy/image.py [hunk 5]
```diff
@@ -1,117 +0,0 @@
-# -*- coding: utf-8 -*-
-# ------------------------------------------------------------------------------
```
Comment (DavidMinarsch) [inline autonomy/deploy/image.py:2]: why?

File: autonomy/data/Dockerfiles/agent/Pipfile [hunk 6]
```diff
@@ -1,16 +0,0 @@
-[[source]]
-url = "https://pypi.org/simple"
```
Comment (DavidMinarsch) [inline autonomy/data/Dockerfiles/agent/Pipfile:2]: why removed?

File: autonomy/data/Dockerfiles/agent/start.sh [hunk 7]
```diff
@@ -1,54 +0,0 @@
-#! /bin/bash
-if [ "$DEBUG" == "1" ];
```
Comment (DavidMinarsch) [inline autonomy/data/Dockerfiles/agent/start.sh:2]: why removed?

File: autonomy/cli/utils/click_utils.py [hunk 8]
```diff
@@ -25,27 +25,6 @@
 
 from autonomy.analyse.abci.app_spec import DFA
 from autonomy.deploy.chain import CHAIN_CONFIG
-from autonomy.deploy.image import ImageProfiles
-
-
-def image_profile_flag(
```
Comment (DavidMinarsch) [inline autonomy/cli/utils/click_utils.py:31]: ?

**Comment by angrybayblade** [comment]
> Ugh why are you moving things back into a deployments folder?

As I described in the doc, building images when running deployments is not a good idea. We actually don't need to build images everytime we want to build or run a deployment. This PR addresses that. From now on we build images with open-autonomy releases and users can just pull and run them when running deployments.

File: .github/workflows/main_workflow.yml [hunk 9]
```diff
@@ -196,18 +196,17 @@ jobs:
...
-          pip install .
-          autonomy fetch --local --service valory/hello_world
-          
-          cd hello_world
-          autonomy build-images
-          autonomy build-images --dev
-          autonomy build-images --dependencies
-          autonomy build-images --dependencies --version 1.0.0
-          autonomy deploy build ../deployments/keys/hardhat_keys.json
+          # build default images
```
Comment (DavidMinarsch) [inline .github/workflows/main_workflow.yml:199]: don't follow, why are we regressing?

File: deployments/Dockerfiles/autonomy/Dockerfile (hunk 4 above)
Comment (angrybayblade) [reply deployments/Dockerfiles/autonomy/Dockerfile:2]: They were constants only used once, so I just replaced them with actual values where they were used.

**Comment by DavidMinarsch** [comment]
> > Ugh why are you moving things back into a deployments folder?
> 
> As I described in the doc, building images when running deployments is not a good idea. We actually don't need to build images everytime we want to build or run a deployment. This PR addresses that. From now on we build images with open-autonomy releases and users can just pull and run them when running deployments.

No you misunderstood. I agree that it is a good idea to provide base images. But the user always has to build the agent specific image of course! Otherwise, the deployment will take very long as the agent deps have to be downloaded, installed and built...

File: autonomy/deploy/image.py (hunk 5 above)
Comment (angrybayblade) [reply autonomy/deploy/image.py:2]: https://github.com/valory-xyz/open-autonomy/pull/1170#issuecomment-1206577391

**Comment by angrybayblade** [comment]
The fixes proposed here have been addressed properly on #1218 and #1221

**Pull request closed without merging**